tests/qinfo/test_entropies.py

```diff
@@ -309,3 +315,198 @@ def circuit_state(x):
         grad_expected_entropy = expected_entropy_grad_ising_xx(param) / np.log(base)
 
         assert qml.math.allclose(grad_entropy, grad_expected_entropy, rtol=1e-04, atol=1e-05)
+
+
+class TestMutualInformation:
+    """Tests for the mutual information functions"""
+
+    @pytest.mark.parametrize("device", ["default.qubit", "default.mixed"])
+    @pytest.mark.parametrize("interface", ["autograd", "jax", "tensorflow", "torch"])
+    @pytest.mark.parametrize(
+        "params", [np.array([0.0, 0.0]), np.array([0.3, 0.4]), np.array([0.6, 0.8])]
+    )
+    def test_qnode_state(self, device, interface, params):
+        """Test that mutual information works for QNodes that return the state"""
+        dev = qml.device(device, wires=2)
+
+        params = qml.math.asarray(params, like=interface)
+
+        @qml.qnode(dev, interface=interface)
+        def circuit(params):
+            qml.RY(params[0], wires=0)
+            qml.RY(params[1], wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.state()
+
+        actual = qml.qinfo.mutual_info_transform(circuit, indices0=[0], indices1=[1])(params)
+
+        # compare QNode results with the results of computing directly from the state
+        state = circuit(params)
+        expected = qml.math.to_mutual_info(state, indices0=[0], indices1=[1])
```

Comment:
Here I would avoid using this function to calculate the mutual information but rather have analytical values that you derived yourself or find in the literature. Alternatively you can keep the test and add another one for that purpose.

Reply:
Changed to compare against analytic values.